v2 npm: blockers for webpack/Next.js consumers (packaging + conditional exports)

**Context**: built a Next.js frontend page that creates token contracts via `dash-platform-sdk@1.4.0-dev.11` (which pulls `pshenmic-dpp@2.0.0-dev.16`). End-to-end works after two browser-bundling workarounds applied. Both look like they'd hit any webpack/Next.js consumer of v2.

---

### 1. `wasmCreation.cjs` requires `.js` but tarball ships `.cjs`

**Source**: https://github.com/owl352/pshenmic-dpp/blob/v2.0.0-dev/dist/binaries/wasmCreation.cjs#L6

```js
const { decode } = require('../utils/base122.js');
```

The tarball contains `dist/utils/base122.cjs`, not `.js`. Node's CJS resolver may forgive in some setups; webpack does not, breaks Next.js compile with `Module not found: Can't resolve '../utils/base122.js'`.

**Fix**: change `.js` to `.cjs` (one character).

**Workaround in our PR**: [`patches/pshenmic-dpp+2.0.0-dev.16.patch`](https://github.com/pshenmic/platform-explorer/pull/795).

---

### 2. `"node"` conditional export pulls NAPI-native build into browser bundle

**Source**: https://github.com/owl352/pshenmic-dpp/blob/v2.0.0-dev/package.json — `exports`:

```json
".": {
  "node": "./dist/src/native.js",
  "react-native": "./dist/src/react-native.js",
  "default": "./dist/src/wasm.js"
}
```

Next.js SSR runs in Node so the `"node"` condition matches; webpack then tries to bundle `native.js` for the browser, follows through `node.cjs` → `bindingsTypes.d.ts` (a TypeScript declarations file), and explodes with `Module parse failed: Unexpected token (10:7) export declare class ActionTakerNAPI`.

**Fix options**:
- Drop the `"node"` condition (let Node fall through to `default` = wasm)
- Or add a `"browser"` condition pinned to `wasm.js` and document that bundlers should set `resolve.conditionNames = ['browser', …]`
- Or document the workaround so consumers don't have to discover it.

**Workaround in our PR**: webpack alias `'pshenmic-dpp$': 'pshenmic-dpp/wasm'` in `next.config.mjs`.

---

### Why not caught earlier

Both bugs only surface in webpack/Next.js builds — invisible from Node-only test runs. We appear to be the first frontend consumer of v2 with a token-creation flow. Working PR with both workarounds + real testnet token (`tMp3naNHBiGNHsQHmABthrCteBuvGqV6NBu5yqok6Cy`): https://github.com/pshenmic/platform-explorer/pull/795

Once both points land in a release, our PR can drop the patch + alias in one chore commit.

--- a/package.json
+++ b/package.json
@@ -29,7 +29,6 @@
   ],
   "exports": {
     ".": {
-      "node": "./dist/src/native.js",
       "react-native": "./dist/src/react-native.js",
       "default": "./dist/src/wasm.js"
     },

--- a/package.json
+++ b/package.json
@@ -29,6 +29,8 @@
   ],
   "exports": {
     ".": {
+      "browser": "./dist/src/wasm.js",
+      "node": "./dist/src/native.js",
       "react-native": "./dist/src/react-native.js",
       "default": "./dist/src/wasm.js"
     },

--- a/templates/binaries/wasmCreation.cjs
+++ b/templates/binaries/wasmCreation.cjs
@@ -2,7 +2,7 @@ const { WASI } = require('@tybys/wasm-util');
 const { getDefaultContext } = require('@emnapi/runtime');
 const { instantiateNapiModuleSync } = require('@emnapi/core');
 const { bytes } = require('./wasm/wasmBytes.cjs');
-const { decode } = require('../utils/base122.js');
+const { decode } = require('../utils/base122.cjs');
 const { decompressSync } = require('fflate');
 
 const wasmBytes = new Uint8Array(decode(bytes));

